### DISCUSSION TITLE
next.config.js redirects and catch all pages

### DISCUSSION THREAD

**Discussion by jaybytez** [vercel/next.js#17906] [Help]
I have a redirect for taking a route and redirecting it to an external page.

Problem is at the base of my project, i have a catch all page [...slug].js and because of this, it appears that the redirect never occurs because the requested route (that should redirect) enters the catch all page instead of being redirected and throws an error in my getStaticProps.

Do redirects in the next.config.js not occur first before routing urls to pages?   Is there a way to force those redirects to occur before routing to pages?

**Comment by AldoS** [comment]
# Bug report

## Describe the bug

I can confirm this is happening also in my project.
When we land directly on a page, that should redirect, the redirection works fine.
The problem is when you navigate through pages. The catch-all is taking place instead of redirecting

## To Reproduce

1. Add a catch-all route, e.g. pages/[[...alias]].js
1. Update `next.config.js` with redirects as described in docs https://nextjs.org/docs/api-reference/next.config.js/redirects
1. Build the project
1. Load a page and then navigate to a page that redirection should take place
1. Redirection is not fired, instead, catch-all routes catch it

## Expected behavior

Redirection to take place before the catch-all routes.

## System information

- OS: [macOS]
- Browser [chrome]
- Version of Next.js: [9.5.3]
- Version of Node.js: [11.10.1]

**Comment by alexandersch** [comment]
Same here. We need to permanently redirect old URL's but now this is not possible with a catch all in the root...

**Reply by SimeonGriggs** [reply]
I'm also having this issue. 

Seems to be no way to have redirects in `vercel.json` take precedence over `[[...slug]].js` in the pages root.

**Comment by viperfx07** [comment]
Still happens on Next.js 12 and just know that from having a middlware. I'm not sure whether it only happens in SSG case or others too.
Say I've got a redirect from /english to /learning-areas/english-k-10 and a <Link href="/english">. Also got a [[...slug]].js as catch route

1. The middleware's first request.nextUrl `pathname` actually says **/learning-areas/english**
2. However right after that it loads `pathname` = '/' with slug param equals to english
